### PR TITLE
refactor: convert-header-file-to-typescript-and-add-type-checks

### LINKED ISSUE #13585: Add TypeScript types to Header

### PR DESCRIPTION
**Pull request by Cris-Chan**
Closes #13585 

converted header to .tsx and removed prop types replacing them with typeSCript interface. mimicked enforcement of AriaLabelPropType as well (not sure how we want to go about that but made a first pass)
added contributer 

#### Changelog

**New**

- {{new thing}}

**Changed**

- {{Header.js}}

**Removed**

- {{removed thing}}

#### Testing / Reviewing

Ran in story book to make sure no errors were thrown! Place in stroybook page to make sure no errors are thrown!

### CODE REVIEW

**Comment by Cris-Chan** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/UIShell/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const prefix = usePrefix();
+  const className = cx(`${prefix}--header`, customClassName);
+
+  return (
+    <header {...rest} className={className}>
+      {children}
+    </header>
+  );
+};
+
```
Comment (guidari) [inline packages/react/src/components/UIShell/Header.tsx]: We should keep the propTypes for documentation reasons in storybook and for JavaScript users also.

```suggestion
Header.propTypes = {
  /**
   * Required props for the accessibility label of the header
   */
  ...AriaLabelPropType,

  /**
   * Optionally provide a custom class name that is applied to the underlying <header>
   */
  className: PropTypes.string,
};
```

**Review by guidari — CHANGES_REQUESTED** [review]
Hey @Cris-Chan thanks for the contribution!! 🚀

Just one minor change.

File: packages/react/src/components/UIShell/Header.tsx (hunk 1 above)
Comment (Cris-Chan) [reply packages/react/src/components/UIShell/Header.tsx]: oops! think i read that at totally forgot!
committing the proptypes now!

**Review by Cris-Chan — COMMENTED** [review]
prop types added back in, needed for storybook and javascript users

**Review by guidari — APPROVED** [review]
LGTM! 🚀

**Review by andreancardona — APPROVED** [review]
Looks good to me :)

**Comment by andreancardona** [comment]
@Cris-Chan Hey Cris - it looks like we have some failing tests here :/ do you mind fixing those and we can re-review after. Thanks!

**Pull request closed without merging**